src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1144,11 +1221,53 @@ nns_edge_connect (nns_edge_h edge_h, nns_edge_protocol_e protocol,
   }
 
   eh->protocol = protocol;
+  if (0 != g_ascii_strcasecmp (eh->topic, TOPIC_FOR_TCP_CONN)) {
+    gchar *topic, *msg = NULL;
+
+    g_free (eh->ip);
+    eh->ip = g_strdup (ip);
+    eh->port = port;
+    if (!nns_edge_mqtt_is_connected (eh)) {
+      if (NNS_EDGE_ERROR_NONE != nns_edge_mqtt_connect (eh)) {
+        nns_edge_loge ("Connection failure to broker.");
+        nns_edge_unlock (eh);
+        return NNS_EDGE_ERROR_CONNECTION_FAILURE;
+      }
+      topic = g_strdup_printf ("edge/inference/+/%s/#", eh->topic);
+      g_free (eh->topic);
+      eh->topic = topic;
+
+      if (NNS_EDGE_ERROR_NONE != nns_edge_mqtt_subscribe (eh)) {
+        nns_edge_loge ("Failed to subscribe to topic: %s.", eh->topic);
+        nns_edge_unlock (eh);
+        return NNS_EDGE_ERROR_CONNECTION_FAILURE;
+      }
+    }
+
+    if (NNS_EDGE_ERROR_NONE == nns_edge_mqtt_get_message (eh, &msg)) {
+      gchar **splits;
+      splits = g_strsplit (msg, ":", -1);
+      server_ip = g_strdup (splits[0]);
+      server_port = g_ascii_strtoull (splits[1], NULL, 10);
+      nns_edge_logd ("[DEBUG] Parsed server info: Server [%s:%d] ", server_ip,
+          server_port);
+
+      g_strfreev (splits);
+      g_free (msg);
+    } else {
+      nns_edge_loge ("Failed to get server IP addr within timeout.");
+      nns_edge_unlock (eh);
+      return NNS_EDGE_ERROR_CONNECTION_FAILURE;
+    }
+  } else { /** case for NNS_EDGE_PROTOCOL_TCP == eh->protocol */
+    server_ip = g_strdup (ip);
```

Comment:
let's use defined functions in nns-edge repo 
g_strdup -> nns_edge_strdup 
g_free -> nns_edge_free
g_ascii_strtoull > strtoll

Reply:
Changed. Thanks.